src/routes/safe/store/middleware/notificationsMiddleware.js

```diff
@@ -24,16 +27,22 @@ const notificationsMiddleware = (store: Store<GlobalState>) => (next: Function)
         const transactionsList = action.payload
         const userAddress: string = userAccountSelector(state)
         const awaitingTransactions = getAwaitingTransactions(transactionsList, userAddress)
+        const safes = safesMapSelector(state)
+        const safeAddress = safeParamAddressFromStateSelector(state)
+        const currentSafe = safes.get(safeAddress)
 
+        // If the user is not an owner, do not send notifications
+        if (!isUserOwner(currentSafe, userAddress)) {
+          break
+        }
 
-        awaitingTransactions.map((awaitingTransactionsList, safeAddress) => {
-          const convertedList = awaitingTransactionsList.toJS()
-          const notificationKey = `${safeAddress}-${userAddress}`
+        awaitingTransactions.map((awaitingTransactionsList: List<Transaction>, txSafeAddress) => {
+          const notificationKey = `${txSafeAddress}-${userAddress}`
           const onNotificationClicked = () => {
             dispatch(closeSnackbarAction({ key: notificationKey }))
-            dispatch(push(`/safes/${safeAddress}/transactions`))
+            dispatch(push(`/safes/${txSafeAddress}/transactions`))
           }
-          if (convertedList.length > 0) {
+          if (awaitingTransactionsList.size > 0) {
```

Comment:
Is it also possible to check if the size of awaitingTransactionsList is greater than 0 before entering .map and initializing the callbacks?